### DISCUSSION TITLE
classes/class_editorinspectorplugin

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#138] [User-contributed notes]
# classes/class_editorinspectorplugin

Inherits: RefCounted< Object Plugin for adding custom property editors on the inspector. Description: EditorInspectorPlugin allows adding custom property editors to EditorInspector. When an object ...

https://docs.godotengine.org/en/stable/classes/class_editorinspectorplugin.html

**Comment by NateAustin** [comment]
Note that when using _can_handle(), _parse_begin(), etc. in C#, the object will only be correctly identifiable if it is marked with [Tool]. Otherwise it will show as the base node type.

**Comment by OS-of-S** [comment]
Found a bug connected with add_property_editor, stay informed: https://github.com/godotengine/godot/issues/112122
